### PR TITLE
hold on some warnings of resource leak

### PR DESCRIPTION
**Pull request by adyliu**
* fixed warnings of eclipse compiler

### CODE REVIEW

File: ethereumj-core/src/main/java/org/ethereum/net/rlpx/Handshaker.java [hunk 1]
```diff
@@ -78,53 +78,45 @@ public Handshaker() {
...
-        conn.writeMessage(new DisconnectMessage(ReasonCode.PEER_QUITING));
-        conn.handleNextMessage();
-
-        while (true) {
-            try {
-                conn.handleNextMessage();
-            } catch (EOFException e) {
-                break;
+        try (Socket sock = new Socket(host, port)) {
+            InputStream inp = sock.getInputStream();
```
Comment (zilm13) [inline ethereumj-core/src/main/java/org/ethereum/net/rlpx/Handshaker.java]: This way we could put input and output streams in `try` as resources too

File: ethereumj-core/src/main/java/org/ethereum/solidity/compiler/Solc.java [hunk 2]
```diff
@@ -60,18 +60,19 @@ private void initBundled() throws IOException {
...
-        while (scanner.hasNext()) {
-            String s = scanner.next();
-            File targetFile = new File(tmpDir, s);
-            InputStream fis = getClass().getResourceAsStream("/native/" + getOS() + "/solc/" + s);
-            Files.copy(fis, targetFile.toPath(), StandardCopyOption.REPLACE_EXISTING);
-            if (solc == null) {
-                // first file in the list denotes executable
-                solc = targetFile;
-                solc.setExecutable(true);
+        try(Scanner scanner = new Scanner(is)){
```
Comment (zilm13) [inline ethereumj-core/src/main/java/org/ethereum/solidity/compiler/Solc.java]: Spaces, please
after `try` and before `{`

File: ethereumj-core/src/main/java/org/ethereum/sync/FastSyncManager.java [hunk 3]
```diff
@@ -716,7 +716,8 @@ public void main() {
                         }
 
                         syncUnsecure(pivot);  // regularSync should be inited here
-                    case SECURE:
+                    //$FALL-THROUGH$
+                case SECURE:
```
Comment (zilm13) [inline ethereumj-core/src/main/java/org/ethereum/sync/FastSyncManager.java]: you've broken spacing

File: ethereumj-core/src/main/java/org/ethereum/sync/FastSyncManager.java [hunk 4]
```diff
@@ -726,7 +727,8 @@ public void main() {
                         syncSecure();
 
                         fireSyncDone(SECURE);
-                    case COMPLETE:
+                    //$FALL-THROUGH$
+                case COMPLETE:
```
Comment (zilm13) [inline ethereumj-core/src/main/java/org/ethereum/sync/FastSyncManager.java]: you've broken spacing

File: ethereumj-core/src/main/java/org/ethereum/sync/FastSyncManager.java [hunk 5]
```diff
@@ -716,7 +716,8 @@ public void main() {
                         }
 
                         syncUnsecure(pivot);  // regularSync should be inited here
-                    case SECURE:
+                    //$FALL-THROUGH$
```
Comment (zilm13) [inline ethereumj-core/src/main/java/org/ethereum/sync/FastSyncManager.java]: It's very IDE-specific, it's not standard, I'd not use it
And one down here

File: ethereumj-core/src/test/java/org/ethereum/core/ImportLightTest.java [hunk 6]
```diff
@@ -79,15 +79,16 @@ public void importBlocks() throws Exception {
...
         BlockchainImpl blockchain = createBlockchain(GenesisLoader.loadGenesis(
                 getClass().getResourceAsStream("/genesis/frontier.json")));
-        Scanner scanner = new Scanner(new FileInputStream("D:\\ws\\ethereumj\\work\\blocks-rec.dmp"));
-        while (scanner.hasNext()) {
-            String blockHex = scanner.next();
-            Block block = new Block(Hex.decode(blockHex));
-            ImportResult result = blockchain.tryToConnect(block);
-            if (result != ImportResult.EXIST && result != ImportResult.IMPORTED_BEST) {
-                throw new RuntimeException(result + ": " + block + "");
+        try(Scanner scanner = new Scanner(new FileInputStream("D:\\ws\\ethereumj\\work\\blocks-rec.dmp"))){
```
Comment (zilm13) [inline ethereumj-core/src/test/java/org/ethereum/core/ImportLightTest.java]: Spaces!

File: ethereumj-core/src/test/java/org/ethereum/net/UdpTest.java [hunk 7]
```diff
@@ -202,8 +202,7 @@ public void client() throws Exception {
 
     public static String bindIp() {
         String bindIp;
-            try {
-                Socket s = new Socket("www.google.com", 80);
+            try (Socket s = new Socket("www.google.com", 80)){
```
Comment (zilm13) [inline ethereumj-core/src/test/java/org/ethereum/net/UdpTest.java]: Space before `{`

**Comment by mkalinin** [comment]
@adyliu any progress in requested changes?

**Pull request merged**